app/src/main/java/com/android/streetworkapp/ui/navigation/InfoDialogs.kt

```diff
@@ -0,0 +1,138 @@
+                  "This is the park overview, here you can see the park's details and the events that are planned in it. \n\n You can create an event or join an already existing one. You can also rate the park !"),
+          screens.ADD_EVENT to
+              InfoDialog(
+                  "AddEvent",
+                  "How to create an event ?",
+                  "This is your firsts steps towards creating an event ! \n\n You simply have to add a cool title, then, describe your event : What are you going to do there ? \n\n Finally, set the date and time and you are good to go !"),
+          screens.EVENT_OVERVIEW to
+              InfoDialog(
+                  "EventOverview",
+                  "What is the Event Overview ?",
+                  "This is the event overview, here you can access all of the information that you need. \n\n You are free to join the event if you want to participate !"),
+          screens.PROGRESSION to
+              InfoDialog(
+                  "Progression",
+                  "What is the Progression ?",
+                  "This is the progression screen, here you can see your progression and achievements. \n\n There is your record for each exercise in Training, click on it to see more details !")
+
+          // Auth not done because unnecessary
+          // Add more if needed
+          )
+
+  /** Returns the default InfoDialog (for screens that are not supported). */
+  fun defaultInfoDialog(): InfoDialog {
+    return InfoDialog(
+        "Default",
+        "Screen not supported yet",
+        "Information about this screen is not available. Please explore to learn more!")
+  }
+
+  /**
+   * This function sets up the INFO action callback for the dialog. Cannot be done in another way
+   * because it needs the topAppBarManager to be instantiated.
+   */
+  fun setUp() {
+    Log.d("InfoDialog", "Setting up actions callbacks with Manager $topAppBarManager")
+    topAppBarManager!!.setActionCallback(TopAppBarManager.TopAppBarAction.INFO) {
+      Log.d("InfoDialog", "Info callback start : show = ${showDialog.value}")
+      showDialog.value = true
+      Log.d("InfoDialog", "Info callback end : show = ${showDialog.value}")
+    }
+  }
+
+  /**
+   * This function displays the info dialog for the current screen. If the screen is not supported,
+   * a default "non-supported" version is displayed.
+   */
+  @Composable
+  fun Display() {
+    Log.d("InfoDialog", "Displaying info dialog ${showDialog.value} for screen ${screenName.value}")
+    val func = infoDialogs[screenName.value]
+    if (func != null) {
+      Log.d("InfoDialog", "Displaying info dialog : ${func.title}")
+      func.DisplayInfoDialog(showDialog)
+    } else {
+      Log.d("InfoDialog", "Displaying default info dialog : ${defaultInfoDialog().title}")
+      defaultInfoDialog().DisplayInfoDialog(showDialog)
+    }
+  }
+
+  /** Returns the InfoDialogs mapping */
+  fun getInfoDialogs(): MutableMap<String, InfoDialog> {
```

Comment:
Suggestion: as a convention, it is always better to use Kotlin implicit getters instead of creating your own getter function:
```
// Mapping Screen to InfoDialog
  private val _infoDialogs: MutableMap<String, InfoDialog> = ...
    val infoDialogs: MutableMap<String, InfoDialog> 
        get() = _infoDialogs
```